src/module.c

```diff
@@ -5864,6 +5912,13 @@ robj **moduleCreateArgvFromUserFormat(const char *cmdname, const char *fmt, int
  *              If everything succeeded, it will return with a NULL, otherwise it will
  *              return with a CallReply object denoting the error, as if it was called with
  *              the 'E' code.
+ *     * 'K' -- Argument is a callback to call when client gets unblock (in case of blocking command)
+ *              follow private data (void*) that will be given to the callback. In case the command
+ *              did not block the client, the code flow stays the same. In case the command
+ *              did block the client, the given callback will be called when the client gets unblock.
+ *              callback signature is define on 'RedisModuleOnUnblocked', its gets the RedisModuleCtx,
+ *              the command reply, and the private data that was give when calling 'RM_Call'.
+ *              The command reply is of type RedisModuleReply and need to be freed by the user.
```

Comment:
there are some grammar issues here, but regardless, as i noted earlier, i'm not sure i like this approach, starting a thread here to discuss the alternative.

Reply:
@yossigo @guybe7 ?

Reply:
@oranagra I have updated the code and the top comment with the API changes we discussed (RM_Call returns a promise object that allow to set the unblock handler). Let me know what you think.